h5py/_hl/files.py

```diff
@@ -502,16 +506,17 @@ def __init__(self, name, mode='r', driver=None, libver=None, userblock_size=None
         """
         if driver == 'ros3':
             if ros3:
-                from urllib.parse import urlparse
-                url = urlparse(name)
-                if url.scheme == 's3':
-                    aws_region = kwds.get('aws_region', b'').decode('ascii')
-                    if len(aws_region) == 0:
-                        raise ValueError('AWS region required for s3:// location')
-                    name = f'https://s3.{aws_region}.amazonaws.com/{url.netloc}{url.path}'
-                elif url.scheme not in ('https', 'http'):
-                    raise ValueError(f'{name}: S3 location must begin with '
-                                     'either "https://", "http://", or "s3://"')
+                if hdf5_version < (2, 0, 0):
```

Comment:
while we're on this, maybe it's a good time to also rewrite the `else` branch as a guard and save an indentation level. Something like
```py
if driver == 'ros3':
    if not ros3:
        raise ValueError(...)
    # business logic goes here
```

Reply:
Fixed in 0504312.